// vocs/_lib/vite/.vocs/dist/assets/dynamic-Qg3PdJ3t.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Dynamic Integration",
  "description": "Add your Dynamic-powered app to Intersend.",
  "slug": "/dynamic"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h2: "h2",
    h3: "h3",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsxs(_components.h2, {
      id: "prerequisite",
      children: ["Prerequisite", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#prerequisite",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Your existing web app is powered by ", jsx(_components.a, {
          href: "https://dynamic.xyz",
          children: "Dynamic"
        }), " for authentication and embedded wallet management."]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "integration-steps",
      children: ["Integration Steps", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-steps",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "1-install-intersend-evm-wallet-connector",
      children: ["1. Install Intersend EVM Wallet Connector", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-install-intersend-evm-wallet-connector",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Install the Dynamic Intersend EVM wallet connector with the following command:"
    }), "\n", jsx(_components.p, {
      children: "For Dynamic v3:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npm"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " install"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " @dynamic-labs-connectors/intersend-evm@3.2.4"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# or"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "yarn"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " add"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " @dynamic-labs-connectors/intersend-evm@3.2.4"
            })]
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "For Dynamic v4:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npm"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " install"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " @dynamic-labs-connectors/intersend-evm@4.0.0"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# or"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "yarn"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " add"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " @dynamic-labs-connectors/intersend-evm@4.0.0"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "2-import-the-intersend-connector",
      children: ["2. Import the Intersend Connector", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-import-the-intersend-connector",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "In your project, import the IntersendEvmWalletConnectors:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsx(_components.code, {
          children: jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "import"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " { IntersendEvmWalletConnectors } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "from"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " '@dynamic-labs-connectors/intersend-evm'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ";"
            })]
          })
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "3-configure-dynamic-provider",
      children: ["3. Configure Dynamic Provider", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#3-configure-dynamic-provider",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Add the Intersend EVM wallet connector to your DynamicContextProvider:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: "DynamicContextProvider"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  settings"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    walletConnectors: [EthereumWalletConnectors, IntersendEvmWalletConnectors],"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "  {"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "/* Your app code here */"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "</"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: "DynamicContextProvider"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "4-whitelist-intersend-domain",
      children: ["4. Whitelist Intersend Domain", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#4-whitelist-intersend-domain",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Add Intersend domains to your Dynamic dashboard's allowed origins:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "https://app.intersend.io",
          children: "https://app.intersend.io"
        })
      }), "\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "https://intersend.io",
          children: "https://intersend.io"
        })
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.a, {
        href: "https://docs.dynamic.xyz/developer-dashboard/security#adding-an-origin",
        children: "Configure Allowed Origins →"
      })
    }), "\n", jsxs(_components.h3, {
      id: "5-verify-app-behavior-inside-intersend",
      children: ["5. Verify App Behavior Inside Intersend", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#5-verify-app-behavior-inside-intersend",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsxs(_components.strong, {
          children: ["Open ", jsx(_components.a, {
            href: "https://app.intersend.io/apps/sandbox",
            children: "Sandbox"
          })]
        }), " and paste your web app’s URL into the address bar of the Intersend browser."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Auto-connect"
        }), " should kick in; Intersend will inject the user’s wallet data into your app."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Test transaction or signing flows"
        }), " to confirm everything relays properly."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "6-get-listed",
      children: ["6. Get Listed", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#6-get-listed",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Once integrated, submit your app for review:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Email: ", jsx(_components.a, {
          href: "mailto:hello@intersend.io",
          children: "hello@intersend.io"
        })]
      }), "\n", jsx(_components.li, {
        children: "Telegram: @erturkarda"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "need-help",
      children: ["Need Help?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#need-help",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: jsx(_components.a, {
          href: "mailto:hello@intersend.io",
          children: "Contact Intersend Support"
        })
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
