// vocs/_lib/vite/.vocs/dist/assets/index-pQ5ZyAzS.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "layout": "minimal",
  "redirect": "/introduction",
  "title": "Intersend Documentation",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "intersend-documentation",
        children: ["Intersend Documentation", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#intersend-documentation",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "You will be redirected to the introduction page momentarily."
    }), "\n", jsxs(_components.p, {
      children: ["If you are not redirected automatically, ", jsx(_components.a, {
        href: "/introduction",
        children: "click here"
      }), "."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
