// vocs/_lib/vite/.vocs/dist/assets/introduction-D5wgiEDA.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Introduction",
  "description": "Welcome to the Intersend documentation",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    p: "p",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "welcome-to-intersend",
        children: ["Welcome to Intersend", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#welcome-to-intersend",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Intersend provides powerful tools for building web3 applications and managing blockchain interactions. This documentation will help you get started quickly and provide in-depth information on all of our features and APIs."
    }), "\n", jsxs(_components.h2, {
      id: "getting-started",
      children: ["Getting Started", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#getting-started",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The first step to building with Intersend is understanding the core concepts and setting up your development environment."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/overview",
          children: "Overview"
        }), " - Learn about Intersend's architecture and capabilities"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/quickstart/create",
          children: "Quickstart"
        }), " - Build your first application with Intersend"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/api-reference/introduction",
          children: "API Reference"
        }), " - Explore our API endpoints"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "core-features",
      children: ["Core Features", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#core-features",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Intersend offers a comprehensive set of tools for web3 developers:"
    }), "\n", jsxs(_components.h3, {
      id: "authentication--integration",
      children: ["Authentication & Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#authentication--integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/auth-providers/overview-auth",
          children: "Auth Providers"
        }), " - Web3 authentication solutions"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/walletconnect",
          children: "WalletConnect Integration"
        }), " - Connect with WalletConnect"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/rainbowkit",
          children: "RainbowKit"
        }), " - Beautiful wallet connection experiences"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "application-building",
      children: ["Application Building", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#application-building",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/apps/overview",
          children: "Apps Overview"
        }), " - Application integration options"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/waas/overview-waas",
          children: "Wallet-as-a-Service"
        }), " - Custom wallet solutions"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "development-resources",
      children: ["Development Resources", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#development-resources",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/essentials/code",
          children: "Code Examples"
        }), " - Sample code and implementation guides"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/development",
          children: "Development Guide"
        }), " - Tools and best practices"]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
