// vocs/_lib/vite/.vocs/dist/assets/introduction-DCNXkG2M.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "API Reference",
  "description": "Introduction to Intersend API endpoints",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    p: "p",
    pre: "pre",
    span: "span",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "api-reference",
        children: ["API Reference", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#api-reference",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Welcome to the Intersend API documentation. This section provides detailed information about our API endpoints, authentication methods, and request/response formats."
    }), "\n", jsxs(_components.h2, {
      id: "authentication",
      children: ["Authentication", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#authentication",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "All API endpoints are authenticated using Bearer tokens. You'll need to include your API key in the Authorization header of your requests:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsx(_components.code, {
          children: jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "Authorization:"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " Bearer"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " YOUR_API_KEY"
            })]
          })
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "base-url",
      children: ["Base URL", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#base-url",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The base URL for all API requests is:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsx(_components.code, {
          children: jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "https://api.intersend.io/v1"
            })
          })
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "endpoints",
      children: ["Endpoints", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#endpoints",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Our API provides the following main endpoint categories:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/api-reference/endpoint/create",
          children: "Create Endpoint"
        }), " - Create new resources"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/api-reference/endpoint/get",
          children: "Get Endpoint"
        }), " - Retrieve resource data"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/api-reference/endpoint/delete",
          children: "Delete Endpoint"
        }), " - Remove resources"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "response-format",
      children: ["Response Format", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#response-format",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "All responses are returned in JSON format and follow a standard structure:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "success"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "true"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "data"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "    // Response data goes here"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  },"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "error"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "null"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "In case of an error, the response will include error details:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "success"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "false"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "data"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "null"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "error"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '    "code"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ERROR_CODE"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '    "message"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Error description"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
