// vocs/_lib/vite/.vocs/dist/assets/navigation-cMFasslC.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Navigation",
  "description": "The navigation field in mint.json defines the pages that go in the navigation menu",
  "icon": "map"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h2: "h2",
    p: "p",
    pre: "pre",
    span: "span",
    ...useMDXComponents(),
    ...props.components
  }, { CodeGroup, Warning } = _components;
  if (!CodeGroup) _missingMdxReference("CodeGroup");
  if (!Warning) _missingMdxReference("Warning");
  return jsxs(Fragment, {
    children: [jsx(_components.p, {
      children: "The navigation menu is the list of links on every website."
    }), "\n", jsxs(_components.p, {
      children: ["You will likely update ", jsx(_components.code, {
        children: "mint.json"
      }), " every time you add a new page. Pages do not show up automatically."]
    }), "\n", jsxs(_components.h2, {
      id: "navigation-syntax",
      children: ["Navigation syntax", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#navigation-syntax",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Our navigation syntax is recursive which means you can make nested navigation groups. You don't need to include ", jsx(_components.code, {
        children: ".mdx"
      }), " in page names."]
    }), "\n", jsxs(CodeGroup, {
      children: [jsx(Fragment, {
        children: jsx(_components.pre, {
          className: "shiki shiki-themes github-light github-dark-dimmed",
          style: {
            backgroundColor: "#fff",
            "--shiki-dark-bg": "#22272e",
            color: "#24292e",
            "--shiki-dark": "#adbac7"
          },
          tabIndex: "0",
          children: jsxs(_components.code, {
            children: [jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"navigation"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": ["
              })]
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "    {"
              })
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '        "group"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": "
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"Getting Started"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ","
              })]
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '        "pages"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": ["
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"quickstart"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "]"
              })]
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "    }"
              })
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "]"
              })
            })]
          })
        })
      }), jsx(Fragment, {
        children: jsx(_components.pre, {
          className: "shiki shiki-themes github-light github-dark-dimmed",
          style: {
            backgroundColor: "#fff",
            "--shiki-dark-bg": "#22272e",
            color: "#24292e",
            "--shiki-dark": "#adbac7"
          },
          tabIndex: "0",
          children: jsxs(_components.code, {
            children: [jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"navigation"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": ["
              })]
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "    {"
              })
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '        "group"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": "
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"Getting Started"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ","
              })]
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '        "pages"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": ["
              })]
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '            "quickstart"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ","
              })]
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "            {"
              })
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '                "group"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": "
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"Nested Reference Pages"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ","
              })]
            }), "\n", jsxs(_components.span, {
              className: "line",
              children: [jsx(_components.span, {
                style: {
                  color: "#005CC5",
                  "--shiki-dark": "#8DDB8C"
                },
                children: '                "pages"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: ": ["
              }), jsx(_components.span, {
                style: {
                  color: "#032F62",
                  "--shiki-dark": "#96D0FF"
                },
                children: '"nested-reference-page"'
              }), jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "]"
              })]
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "            }"
              })
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "        ]"
              })
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "    }"
              })
            }), "\n", jsx(_components.span, {
              className: "line",
              children: jsx(_components.span, {
                style: {
                  color: "#24292E",
                  "--shiki-dark": "#ADBAC7"
                },
                children: "]"
              })
            })]
          })
        })
      })]
    }), "\n", jsxs(_components.h2, {
      id: "folders",
      children: ["Folders", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#folders",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Simply put your MDX files in folders and update the paths in ", jsx(_components.code, {
        children: "mint.json"
      }), "."]
    }), "\n", jsxs(_components.p, {
      children: ["For example, to have a page at ", jsx(_components.code, {
        children: "https://yoursite.com/your-folder/your-page"
      }), " you would make a folder called ", jsx(_components.code, {
        children: "your-folder"
      }), " containing an MDX file called ", jsx(_components.code, {
        children: "your-page.mdx"
      }), "."]
    }), "\n", jsx(Warning, {
      children: jsxs(_components.p, {
        children: ["You cannot use ", jsx(_components.code, {
          children: "api"
        }), " for the name of a folder unless you nest it inside another folder. Mintlify uses Next.js which reserves the top-level ", jsx(_components.code, {
          children: "api"
        }), " folder for internal server calls. A folder name such as ", jsx(_components.code, {
          children: "api-reference"
        }), " would be accepted."]
      })
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"navigation"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    {"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '        "group"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Group Name"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '        "pages"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"your-folder/your-page"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "]"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "]"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "hidden-pages",
      children: ["Hidden Pages", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#hidden-pages",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["MDX files not included in ", jsx(_components.code, {
        children: "mint.json"
      }), " will not show up in the sidebar but are accessible through the search bar and by linking directly to them."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
function _missingMdxReference(id, component) {
  throw new Error("Expected component `" + id + "` to be defined: you likely forgot to import, pass, or provide it.");
}
export {
  MDXContent as default,
  frontmatter
};
