// vocs/_lib/vite/.vocs/dist/assets/overview-auth-DzpOXhQ4.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Overview - For Auth Providers",
  "description": "Make your ecosystem apps instantly embeddable",
  "slug": "/overview-auth"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    br: "br",
    code: "code",
    div: "div",
    em: "em",
    h2: "h2",
    h3: "h3",
    hr: "hr",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  }, { Note } = _components;
  if (!Note) _missingMdxReference("Note");
  return jsxs(Fragment, {
    children: [jsxs(Note, {
      children: [jsxs(_components.p, {
        children: [jsx(_components.strong, {
          children: "This guide is for you if:"
        }), jsx(_components.br, {}), "\n", "You provide authentication/wallet management infrastructure (e.g Privy, Dynamic) and want to:"]
      }), jsxs(_components.ul, {
        children: ["\n", jsx(_components.li, {
          children: "Make client apps portable across Web3 interfaces (Worlcoin, Frames, Telegram, Safe)"
        }), "\n", jsx(_components.li, {
          children: "Offer embeddable apps as a value-add feature"
        }), "\n", jsx(_components.li, {
          children: "Expand ecosystem reach without additional work"
        }), "\n"]
      })]
    }), "\n", jsx(_components.p, {
      children: "Turn every app using your SDK into an embeddable component for Intersend-integrated wallets and platforms."
    }), "\n", jsxs(_components.h2, {
      id: "integration-benefits",
      children: ["Integration Benefits", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-benefits",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Ecosystem Growth"
        }), jsx(_components.br, {}), "\n", "Client apps become instantly available to all Intersend partner wallets"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Zero Client Changes"
        }), jsx(_components.br, {}), "\n", "Add Intersend connector at SDK layer - no app modifications needed"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Universal Compatibility"
        }), jsx(_components.br, {}), "\n", "Works with any partner wallet - making it easy to integrate apps with a few lines of code"]
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "how-it-works",
      children: ["How It Works", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-it-works",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Below is a simplified flow describing how an app (built with your auth SDK) interacts with a user's wallet through Intersend."
    }), "\n", jsxs(_components.h2, {
      id: "example",
      children: ["Example", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#example",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [`A wallet infrastructure provider (let's call this CoinPace Wallet) uses Intersend SDK to add "App X" into their explore page. `, jsx(_components.em, {
            children: "App X is built with your authentication SDK."
          })]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Users click App X within CoinPace Wallet's interface. The app loads in an iframe."
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Your Authentication SDK automatically injects Intersend's connector to App X at initial page load."
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Intersend's iframe middleware facilitates sharing user's account with the app and automatically connects to it with SIWE (Sign in with Ethereum)"
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "When transactions occur:"
        }), "\n", jsxs(_components.ol, {
          children: ["\n", jsx(_components.li, {
            children: "App X constructs transaction data"
          }), "\n", jsx(_components.li, {
            children: "Intersend relays payload to CoinPace Wallet"
          }), "\n", jsx(_components.li, {
            children: "User approves/declines in native wallet UI"
          }), "\n", jsx(_components.li, {
            children: "Result propagates back to the app"
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "end-result",
      children: ["End Result", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#end-result",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The app is fully functional, automatically connected to the user's wallet/account"
      }), "\n", jsx(_components.li, {
        children: "Transactions, signatures, chain switching, etc. flow through Intersend's secure layer"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "flow-diagram",
      children: ["Flow Diagram", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#flow-diagram",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "sequenceDiagram"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant User"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Wallet as Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Intersend as Intersend"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant App as dApp"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,Wallet: Load & Browse Apps"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Opens Explore Tab of the Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Fetch list of available apps"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend-->>Wallet: Returns app list"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,App: Launch Selected App"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Selects an app"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Loads app in iframe"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: Wraps dApp within Intersend interface"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>Wallet: REQUEST_WALLET_INFO"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: WALLET_INFO (address, chainId)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,App: Transaction Flow"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>App: Initiates transaction (e.g. purchase, swap)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    App->>Intersend: eth_sendTransaction (RPC request)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>Wallet: Constructs, and sends tx data"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>User: Prompts user to approve transaction"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Approves"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Wallet: Signs & broadcasts to blockchain"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Returns txHash (RPC response)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: Send success or fail"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    App-->>User: Shows success/fail message"
            })
          })]
        })
      })
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
function _missingMdxReference(id, component) {
  throw new Error("Expected component `" + id + "` to be defined: you likely forgot to import, pass, or provide it.");
}
export {
  MDXContent as default,
  frontmatter
};
