// vocs/_lib/vite/.vocs/dist/assets/overview-DZqLdrGL.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Apps Overview",
  "description": "Introduction to apps integration with Intersend",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "apps-overview",
        children: ["Apps Overview", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#apps-overview",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Intersend provides seamless integration with various crypto applications through our Interspace marketplace. This section will guide you through the different app integrations available and how to use them effectively."
    }), "\n", jsxs(_components.h2, {
      id: "available-app-integrations",
      children: ["Available App Integrations", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#available-app-integrations",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Intersend supports multiple app integrations that expand the functionality of your wallet:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/apps/dynamic",
          children: jsx(_components.strong, {
            children: "Dynamic"
          })
        }), " - Comprehensive identity and authentication solution"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/apps/magic",
          children: jsx(_components.strong, {
            children: "Magic"
          })
        }), " - Passwordless authentication for web3 applications"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/apps/privy",
          children: jsx(_components.strong, {
            children: "Privy"
          })
        }), " - User authentication and wallet management"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.a, {
          href: "/apps/safe",
          children: jsx(_components.strong, {
            children: "Safe"
          })
        }), " - Multi-signature wallet solution"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "integration-benefits",
      children: ["Integration Benefits", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-benefits",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "By integrating with Intersend, apps gain several advantages:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Simplified User Onboarding"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "One-click connection to your app directly from Intersend wallet"
          }), "\n", jsx(_components.li, {
            children: "No complex wallet connection flows"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Enhanced Transaction Experience"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Users can perform transactions without leaving your app"
          }), "\n", jsx(_components.li, {
            children: "Support for multiple blockchain networks in a single integration"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Expanded User Base"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Access to Intersend's growing ecosystem of users"
          }), "\n", jsx(_components.li, {
            children: "Cross-promotion opportunities within the Interspace marketplace"
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "implementation-process",
      children: ["Implementation Process", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#implementation-process",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Integrating your app with Intersend involves:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Register your app in the Intersend developer portal"
      }), "\n", jsx(_components.li, {
        children: "Implement the Intersend SDK in your application"
      }), "\n", jsx(_components.li, {
        children: "Configure the connection settings and transaction flows"
      }), "\n", jsx(_components.li, {
        children: "Test and deploy your integration"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "For detailed implementation guides, select the specific app integration from the sidebar."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
