// vocs/_lib/vite/.vocs/dist/assets/overview-waas-B1L6UlFG.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Overview - For Wallets & WaaS",
  "description": "Embed and manage dApps without maintenance overhead",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    blockquote: "blockquote",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    hr: "hr",
    li: "li",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "wallets--waas-overview",
        children: ["Wallets & WaaS Overview", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#wallets--waas-overview",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.blockquote, {
      children: ["\n", jsx(_components.strong, {
        children: "This guide is for you if:"
      }), "\n", jsx(_components.p, {
        children: "You provide wallet infrastructure (MPC, custodial, non-custodial) (e.g., UniversalX, MetaMask, Phantom) and want to:"
      }), "\n", jsxs(_components.ul, {
        children: ["\n", jsx(_components.li, {
          children: "Offer curated dApps without maintaining it yourself"
        }), "\n", jsx(_components.li, {
          children: "Keep full control over security and UX"
        }), "\n", jsx(_components.li, {
          children: "Automatically integrate new apps added to Intersend"
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "demo",
      children: ["Demo", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#demo",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "See a demo of an Intersend-powered wallet:"
    }), "\n", jsx("iframe", {
      width: "560",
      height: "315",
      src: "https://www.youtube.com/embed/tb8bpSt_qAY?si=CrAPF8gv57AiUIkd",
      title: "YouTube video player",
      frameBorder: "0",
      allow: "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture",
      allowFullScreen: true
    }), "\n", jsx(_components.p, {
      children: jsx(_components.a, {
        href: "https://github.com/intersend/demo-wallet",
        children: "Github Repo"
      })
    }), "\n", jsx(_components.p, {
      children: "Turn your wallet into a Web3 gateway using Intersend's universal connector. Embed dApps directly into your interface while maintaining full control over security and user experience."
    }), "\n", jsxs(_components.h2, {
      id: "why-intersend",
      children: ["Why Intersend?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#why-intersend",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Zero Maintenance"
        }), "\nNew apps become automatically available to your users - no integration work required"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Security First"
        }), "\nAll transactions use your existing signing infrastructure - keys never leave your environment"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Customizable Curation"
        }), "\nChoose which apps to display and control their presentation"]
      }), "\n"]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "how-it-works",
      children: ["How It Works", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-it-works",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Under the hood, Intersend provides an iframe-based connector that mediates communication between your wallet and apps available on Intersend. Your wallet remains in control of signing, transaction approvals, and chain selection while Intersend relays messages securely via ", jsx(_components.code, {
        children: "postMessage"
      }), "."]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "sequenceDiagram"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant User"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Wallet as Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Intersend as Intersend"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant App as dApp"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,Wallet: Load & Browse Apps"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Opens Explore Tab of the Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Fetch list of available apps"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend-->>Wallet: Returns app list"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,App: Launch Selected App"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Selects an app"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Loads app in iframe"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: Wraps the app within Intersend interface"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>Wallet: REQUEST_WALLET_INFO"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: WALLET_INFO (address chainId)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Note over User,App: Transaction Flow"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>App: Initiates transaction (e.g. purchase swap)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    App->>Intersend: eth_sendTransaction (RPC request)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>Wallet: Constructs and sends tx data"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>User: Prompts user to approve transaction"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>Wallet: Approves"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Wallet: Signs & broadcasts to blockchain"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Wallet->>Intersend: Returns txHash (RPC response)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>App: Send success or fail"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    App-->>User: Shows success/fail message"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "security-architecture",
      children: ["Security Architecture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#security-architecture",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The SDK implements a secure messaging protocol between the embedded app and your wallet:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Isolated Iframes:"
          }), "\nAll third-party dApps are sandboxed within an iframe managed by Intersend. This prevents unauthorized access to your code or your users' private keys."]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "No Private Key Exposure:"
          }), "\nAny signature or transaction request is forwarded to your wallet's internal signing logic. Keys never leave your environment."]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsxs(_components.p, {
          children: [jsx(_components.strong, {
            children: "Origin Validation:"
          }), "\nThe Intersend SDK verifies the origin of incoming messages. Suspicious or unknown sources are ignored."]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "contact-us",
      children: ["Contact Us", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contact-us",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Email: ", jsx(_components.a, {
          href: "mailto:hello@intersend.io",
          children: "hello@intersend.io"
        })]
      }), "\n", jsx(_components.li, {
        children: "Telegram: @erturkarda"
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
