// vocs/_lib/vite/.vocs/dist/assets/overview-BTAqsPEG.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Overview",
  "description": "Learn about Intersend platform and its features",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "intersend-overview",
        children: ["Intersend Overview", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#intersend-overview",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Intersend is a marketplace for mini-apps with a smart account for stablecoins. Add money auto-convert to stablecoins. Add accounts and auto-connect to apps. Spend, send, and cash out. Smart profiles for different activities soon."
    }), "\n", jsxs(_components.h2, {
      id: "core-features",
      children: ["Core Features", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#core-features",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "send-and-receive-any-crypto",
      children: ["Send and Receive Any Crypto", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#send-and-receive-any-crypto",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Universal Transactions:"
        }), " Effortlessly send and receive any cryptocurrency across multiple blockchains including EVM chains, Bitcoin, Solana, and more."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Instant Conversion:"
        }), " Receive any crypto directly to your Intersend account as USDT, ensuring stability and simplicity."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "self-custody",
      children: ["Self-Custody", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#self-custody",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Full Ownership:"
        }), " Maintain full control over your assets with a provider-custody wallet."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Enhanced Security:"
        }), " Store your keys locally using passkeys, eliminating the need for secret keys."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "omnichain-support",
      children: ["Omnichain Support", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#omnichain-support",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Extensive Compatibility:"
        }), " Support for thousands of coins and networks, ensuring maximum flexibility and utility."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Real Crypto Apps:"
        }), " Use your crypto like cash with a single click. Don't just hold it; spend it effortlessly."]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "interspace-the-crypto-app-marketplace",
      children: ["Interspace: The Crypto App Marketplace", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#interspace-the-crypto-app-marketplace",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Interspace is Intersend's integrated marketplace for crypto applications designed to simplify user interaction and expand app functionalities."
    }), "\n", jsxs(_components.h3, {
      id: "existing-apps-in-interspace",
      children: ["Existing Apps in Interspace", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#existing-apps-in-interspace",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Bitrefill:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Thousands of gift cards available with a single click. No KYC required."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Trocador:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Obtain Apple/Google Pay-compatible prepaid cards with no KYC."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Onramper:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Instantly buy crypto using hundreds of local payment options with fiat."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "P2P Offramp:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Convert crypto to fiat instantly with low fees and no KYC."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Send Crypto:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Send any crypto across any blockchain including EVM, Bitcoin, Solana, and many others."
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "benefits-for-users",
      children: ["Benefits for Users", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#benefits-for-users",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Easy Wallet Creation:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Create your wallet using social logins, email, existing wallets, or phone."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "No Secret Keys:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Securely store your keys locally using passkeys."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Self-Custody:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Maintain full control of your assets at all times."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Omnichain Support:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Access and utilize thousands of coins and networks including EVM chains, Bitcoin, Solana, Ripple, and more."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Gateway to Real Crypto Apps:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Use your crypto like cash with a single click."
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "benefits-for-apps",
      children: ["Benefits for Apps", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#benefits-for-apps",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "One-Click as a Service:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Users can connect to your app instantly and make transactions with a single click directly from their Intersend wallet."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Flexible Settlements:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Instantly receive any crypto or fiat with a 1% fee from your users."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Customer Acquisition:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Gain access to crypto-native customers, reducing the difficulty of acquiring users in Web3."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Plug-in Tools:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Enhance user experience by offering a variety of apps such as onramp/offramp tools and gift card apps for single-click purchases."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.strong, {
          children: "Universal KYC/KYB:"
        }), "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Users only need to complete KYC once. Encrypted details are shared with your app, streamlining the process and improving UX."
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "contact",
      children: ["Contact", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contact",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Reach out to ", jsx(_components.a, {
        href: "mailto:hello@intersend.io",
        children: "hello@intersend.io"
      }), " or @erturkarda on Telegram."]
    }), "\n", jsxs(_components.h2, {
      id: "get-started-with-intersend",
      children: ["Get Started with Intersend", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#get-started-with-intersend",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Intersend simplifies your crypto journey whether you're managing assets, exploring new apps, or developing innovative solutions. Experience the future of cryptocurrency with Intersend today."
    }), "\n", jsx(_components.p, {
      children: "For more information, visit our website or contact our support team."
    }), "\n", jsx(_components.p, {
      children: "© 2024 Intersend Inc. All Rights Reserved."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
