// vocs/_lib/vite/.vocs/dist/assets/quickstart_technical-BcmJBVYw.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Intersend App Store Technical Documentation",
  "description": "Start building your app on Intersend"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h2: "h2",
    h3: "h3",
    h4: "h4",
    h5: "h5",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsxs(_components.h2, {
      id: "brief-overview",
      children: ["Brief Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#brief-overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Intersend App Store is a non-custodial app marketplace that provides users with modular applications connected through a unified wallet."
    }), "\n", jsx(_components.p, {
      children: "Developers can leverage Intersend APIs, Iframe, and, soon, an SDK to obtain necessary permissions and connect to users' dedicated wallets. Intersend facilitates easy transactions for both Web3 and Web2 users, enabling them to send or receive any cryptocurrency seamlessly and onboard into the ecosystem efficiently."
    }), "\n", jsxs(_components.h2, {
      id: "main-use-cases",
      children: ["Main Use Cases", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#main-use-cases",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Easy Onboarding"
        }), ": Simplifies the process for users to make single-click payments by connecting their wallets."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Unified Wallet Management"
        }), ": Centralized control and management of crypto assets through a single wallet interface."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Omnichain Compatibility"
        }), ": Supports multiple blockchain networks, enabling seamless cross-chain transactions."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Gas-Free Transactions"
        }), ": Allows certain transactions to be conducted without gas fees, improving user experience."]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "ways-to-integrate",
      children: ["Ways to Integrate", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ways-to-integrate",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "1-you-have-an-existing-web-app",
      children: ["1. You have an existing web app", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-you-have-an-existing-web-app",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If you have an existing web application, we can embed your app as an iframe within the Intersend App Store. Your app will communicate with Intersend through a standardized messaging system."
    }), "\n", jsx(_components.p, {
      children: "The iframe sends events, such as making a transaction with details to the deposit address. We capture and process these events on the backend and send the response back to the developer's iframe or web app, which then displays the details to the frontend."
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "sequenceDiagram"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant User"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant YourApp as Your App (iframe)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Intersend as Intersend Backend"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    participant Blockchain"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    User->>YourApp: Accesses App on Intersend Store"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    YourApp->>Intersend: Sends Event (e.g., transaction)"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend->>Blockchain: Processes Event"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Blockchain-->>Intersend: Returns Result"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    Intersend-->>YourApp: Sends Event Result"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    YourApp->>User: Displays Result"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h4, {
      id: "integration-steps",
      children: ["Integration Steps", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-steps",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Prepare Your App: Ensure your app can run within an iframe and handle postMessage communication."
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Install Required Dependencies"
        }), "\n"]
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "npm install viem"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "# or"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "yarn add viem"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.ol, {
      start: "3",
      children: ["\n", jsx(_components.li, {
        children: "Implement Transaction Preparation Function"
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "import"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " { parseEther, encodeFunctionData } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "from"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " 'viem'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ";"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " prepareTransaction"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "to"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "amount"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "tokenAddress"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "  // Encode the function call for a token transfer"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " data"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " encodeFunctionData"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "({"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    abi: [{"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      inputs: ["
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        { name: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'recipient'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", type: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'address'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " },"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        { name: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'amount'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", type: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'uint256'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " }"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      ],"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      name: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'transfer'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      type: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'function'"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }],"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    args: [to, "
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "parseEther"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(amount)]"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  });"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    to: tokenAddress,"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    data,"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    value: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'0'"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: " // Use '0' for token transfers"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  };"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "};"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.ol, {
      start: "4",
      children: ["\n", jsx(_components.li, {
        children: "Implement Message Handling: Set up event listeners for messages from Intersend:"
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "useEffect"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(() "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  const"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " handleMessage"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "event"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " (event.origin "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "!=="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ' "https://app.intersend.io"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ";"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    const"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " { "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "id"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "result"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " event.data;"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "    if"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " (result.success) {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      console."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "log"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'Transaction successful:'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", result.data);"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "      // Update your UI to show success"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    } "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "else"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      console."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "error"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'Transaction failed:'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", result.message);"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "      // Update your UI to show failure"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  };"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  window."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "addEventListener"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'message'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", handleMessage);"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " () "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " window."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "removeEventListener"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'message'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", handleMessage);"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}, []);"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.ol, {
      start: "5",
      children: ["\n", jsx(_components.li, {
        children: "Send Transaction"
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "const"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " sendTransaction"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "recipient"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "amount"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "tokenAddress"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "  const"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " txData"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " prepareTransaction"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(recipient, amount, tokenAddress);"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  window.parent."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "postMessage"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "({"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    method: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'sendTransactions'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    params: {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      txs: [txData],"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      chainId: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'137'"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: " // Replace with your target chain ID"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    },"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    id: "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "`tx-${"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "Date"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "now"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "()"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "}`"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: " // Unique ID for this transaction"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }, "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "'https://app.intersend.io'"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "};"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.ol, {
      start: "6",
      children: ["\n", jsx(_components.li, {
        children: "Update your UI"
      }), "\n"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "// the following snippet is provided as a reference"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "return"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "input"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      type"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"text"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      placeholder"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Recipient Address"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      onChange"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "e"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " setRecipient"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(e.target.value)"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    />"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "input"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      type"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"text"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      placeholder"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Amount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "      onChange"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "("
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "e"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ") "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " setAmount"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(e.target.value)"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    />"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "button"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: " onClick"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "() "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " sendTransaction"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "(recipient, amount, tokenAddress)"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      Send Transaction"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    </"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "button"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "    {"
            }), jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "/* Display transaction status here */"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  </"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.ol, {
      start: "7",
      children: ["\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "The app goes through our testing and approval process."
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["\n", jsx(_components.p, {
          children: "Once approved, your app becomes available to users in the Intersend App Store."
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "2-bring-your-own-apis",
      children: ["2. Bring Your Own APIs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-bring-your-own-apis",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Developers can leverage Intersend's custom-made branded UI by providing their own APIs. This integration method allows you to maintain your backend while taking advantage of Intersend's user-friendly interface and user base."
    }), "\n", jsxs(_components.h4, {
      id: "how-it-works",
      children: ["How It Works", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-it-works",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "You develop and host the required APIs for your app (detailed below)."
      }), "\n", jsx(_components.li, {
        children: "You provide Intersend with your API endpoints and any necessary authentication details."
      }), "\n", jsx(_components.li, {
        children: "Intersend creates a new app within our ecosystem, integrating your APIs with our custom UI."
      }), "\n", jsx(_components.li, {
        children: "The app goes through our testing and approval process."
      }), "\n", jsx(_components.li, {
        children: "Once approved, your app becomes available to users in the Intersend App Store."
      }), "\n"]
    }), "\n", jsxs(_components.h4, {
      id: "required-apis",
      children: ["Required APIs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#required-apis",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "To integrate your app with Intersend, you need to provide the following APIs:"
    }), "\n", jsxs(_components.h5, {
      id: "1-jwt-verification-optional",
      children: ["1. JWT Verification (Optional)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-jwt-verification-optional",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/jwt-verify"
      })]
    }), "\n", jsx(_components.p, {
      children: "This optional endpoint enhances security by verifying the JWT token sent by Intersend."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "token"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "valid"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "true"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "userId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"user123"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "2-app-details",
      children: ["2. App Details", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-app-details",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/details"
      })]
    }), "\n", jsx(_components.p, {
      children: "Provides information about your app for display in the Intersend UI."
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "name"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"YourApp"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "description"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"A brief description of your app"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "logo"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"https://yourapp.com/logo.png"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "status"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"active"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "supportedCurrencies"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"USDT"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "],"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "supportedNetworks"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Bitcoin"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Ethereum"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Tron"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "]"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "3-get-minmax-limits",
      children: ["3. Get Min/Max Limits", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#3-get-minmax-limits",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/get-min-max"
      })]
    }), "\n", jsx(_components.p, {
      children: "Retrieves the minimum and maximum transaction limits for your app."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Bitcoin"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Ethereum"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "min"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0.001"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "max"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"10"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "4-get-exchange-rate",
      children: ["4. Get Exchange Rate", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#4-get-exchange-rate",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/get-rate"
      })]
    }), "\n", jsx(_components.p, {
      children: "Fetches the current exchange rate for a given currency pair."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Bitcoin"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Ethereum"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "amount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "rate"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"15.5"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"15.5"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "5-create-transaction",
      children: ["5. Create Transaction", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#5-create-transaction",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/create-transaction"
      })]
    }), "\n", jsx(_components.p, {
      children: "Initiates a new transaction."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "userId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"user123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Bitcoin"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toNetwork"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Ethereum"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"15.5"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "type"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"swap"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "transactionId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"tx123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "payoutAddress"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"0x1234...5678"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "network"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Bitcoin"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "amount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "userId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"user123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "type"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"swap"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "status"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"pending"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "6-transaction-status",
      children: ["6. Transaction Status", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#6-transaction-status",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/status"
      })]
    }), "\n", jsx(_components.p, {
      children: "Checks the status of a transaction."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "transactionId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"tx123"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "transactionId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"tx123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "status"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"completed"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"15.5"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "timestamp"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"2024-07-15T12:34:56Z"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "7-transaction-history",
      children: ["7. Transaction History", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#7-transaction-history",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/history"
      })]
    }), "\n", jsx(_components.p, {
      children: "Retrieves the transaction history for a user."
    }), "\n", jsx(_components.strong, {
      children: "Request:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "userId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"user123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "page"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "1"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "limit"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "10"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "transactions"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    {"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "transactionId"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"tx123"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "fromAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"1"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "toAmount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"15.5"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "fromCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"BTC"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "toCurrency"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"ETH"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "status"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"completed"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "timestamp"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"2024-07-15T12:34:56Z"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    },"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "    // ... more transactions"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  ],"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "totalCount"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "45"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "currentPage"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "1"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h5, {
      id: "8-faq",
      children: ["8. FAQ", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#8-faq",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Endpoint:"
      }), " ", jsx(_components.code, {
        children: "/faq"
      })]
    }), "\n", jsx(_components.p, {
      children: "Provides frequently asked questions and answers about your app."
    }), "\n", jsx(_components.strong, {
      children: "Response:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "faqs"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": ["
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    {"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "question"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"What is the minimum transaction amount?"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "answer"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: `"The minimum transaction amount varies depending on the cryptocurrency. For BTC, it's 0.001 BTC."`
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    },"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "    // ... more FAQs"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  ]"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "intersend-custom-ui",
      children: ["Intersend Custom UI", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#intersend-custom-ui",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Once you provide these APIs, Intersend will create a custom UI for your app within our ecosystem. Here's an example of how this UI might look:"
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "https://intersend-content-s3-bucket.s3.us-east-1.amazonaws.com/intersend-custom-ui.png",
        alt: "Intersend Custom UI Example"
      })
    }), "\n", jsx(_components.p, {
      children: "Our UI is designed to be intuitive and user-friendly, guiding users through the process of using your app within the Intersend ecosystem. It typically includes sections for:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Selecting currencies and networks"
      }), "\n", jsx(_components.li, {
        children: "Displaying exchange rates"
      }), "\n", jsx(_components.li, {
        children: "Entering transaction amounts"
      }), "\n", jsx(_components.li, {
        children: "Showing transaction status and history"
      }), "\n", jsx(_components.li, {
        children: "Accessing FAQs and support"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "integration-process",
      children: ["Integration Process", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-process",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Develop and host the required APIs on your backend."
      }), "\n", jsx(_components.li, {
        children: "Provide Intersend with your API endpoints and any necessary authentication details."
      }), "\n", jsx(_components.li, {
        children: "Intersend creates a new app in our ecosystem, integrating your APIs with our custom UI."
      }), "\n", jsx(_components.li, {
        children: "We provide you with a test environment to ensure everything works correctly."
      }), "\n", jsx(_components.li, {
        children: "After your approval and our final checks, your app goes live on the Intersend App Store."
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "customization-options",
      children: ["Customization Options", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#customization-options",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "While our standard integration covers most use cases, we understand that some apps may have unique requirements. If you need additional customization, please let us know. We can explore options such as:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Adding app-specific fields to the transaction process"
      }), "\n", jsx(_components.li, {
        children: "Modifying the user flow to better suit your app's requirements"
      }), "\n", jsx(_components.li, {
        children: "Further customizing the UI to match your brand guidelines"
      }), "\n", jsx(_components.li, {
        children: "Integrating additional APIs specific to your app's functionality"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: `By leveraging this "Bring Your Own APIs" integration method, you can maintain control over your backend logic while benefiting from Intersend's user base and intuitive interface. This approach offers a balance between customization and ease of integration, allowing you to quickly bring your app to the Intersend App Store.`
    }), "\n", jsxs(_components.h3, {
      id: "smart-contracts",
      children: ["Smart Contracts", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#smart-contracts",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "We use ERC-4337 smart contracts to facilitate secure and efficient transactions within the Intersend ecosystem."
    }), "\n", jsxs(_components.h2, {
      id: "contact-information",
      children: ["Contact Information", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contact-information",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["For further questions, please reach out to ", jsx(_components.a, {
        href: "mailto:hello@intersend.io",
        children: "hello@intersend.io"
      }), " or @erturkarda on Telegram."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
