// vocs/_lib/vite/.vocs/dist/assets/reusable-snippets-DGwUAtke.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Reusable Snippets",
  "description": "Using reusable content blocks in documentation",
  "layout": "docs",
  "showSidebar": true
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "reusable-snippets",
        children: ["Reusable Snippets", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#reusable-snippets",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Documentation often requires repeating the same content in multiple places. Reusable snippets allow you to define content once and reuse it throughout your documentation."
    }), "\n", jsxs(_components.h2, {
      id: "basic-snippet-usage",
      children: ["Basic Snippet Usage", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#basic-snippet-usage",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Snippets allow you to store reusable content that can be included in multiple pages. This helps maintain consistency and reduces duplication."
    }), "\n", jsxs(_components.h3, {
      id: "creating-snippets",
      children: ["Creating Snippets", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#creating-snippets",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "To create a snippet:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["Add files in the ", jsx(_components.code, {
          children: "docs/snippets"
        }), " directory with your reusable content"]
      }), "\n", jsx(_components.li, {
        children: "Use clear, descriptive names for your snippet files"
      }), "\n", jsx(_components.li, {
        children: "Write your content in MDX format"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["For example, create a file at ", jsx(_components.code, {
        children: "docs/snippets/wallet-setup.mdx"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#005CC5",
                fontWeight: "bold",
                "--shiki-dark": "#6CB6FF",
                "--shiki-dark-font-weight": "bold"
              },
              children: "## Setting Up Your Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "To set up your Intersend wallet:"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "1"
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Download the Intersend app or visit "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "["
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "web.intersend.io"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "]("
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                textDecoration: "underline",
                "--shiki-dark": "#96D0FF",
                "--shiki-dark-text-decoration": "inherit"
              },
              children: "https://web.intersend.io"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: ")"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "2"
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ' Click "Create New Wallet"'
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "3"
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Choose your preferred authentication method"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "4"
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " Follow the on-screen instructions to secure your wallet"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "using-snippets-in-documentation",
      children: ["Using Snippets in Documentation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#using-snippets-in-documentation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "You can then import and use this snippet in any of your documentation pages:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "import"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " WalletSetup "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "from"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " '../snippets/wallet-setup.mdx'"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#005CC5",
                fontWeight: "bold",
                "--shiki-dark": "#6CB6FF",
                "--shiki-dark-font-weight": "bold"
              },
              children: "# Getting Started"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "Before you can start using Intersend, you need to set up your wallet."
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: "WalletSetup"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " />"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "Now that your wallet is set up, let's explore the features..."
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "advanced-snippet-techniques",
      children: ["Advanced Snippet Techniques", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#advanced-snippet-techniques",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "parameterized-snippets",
      children: ["Parameterized Snippets", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#parameterized-snippets",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "You can pass parameters to your snippets for more flexibility:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "export"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " const"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: " ApiEndpoint"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " ="
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ({ "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "method"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "endpoint"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ", "
            }), jsx(_components.span, {
              style: {
                color: "#E36209",
                "--shiki-dark": "#F69D50"
              },
              children: "description"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " }) "
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=>"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " ("
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: " className"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"api-endpoint"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: " className"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"method"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "method"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "</"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: " className"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"endpoint"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "endpoint"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "</"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    <"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: " className"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"description"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "{"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "description"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#F47067"
              },
              children: "}"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "</"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  </"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "div"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ");"
            })
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "Then use it with different values:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: "ApiEndpoint"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  method"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"GET"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  endpoint"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"/api/v1/wallets"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#6CB6FF"
              },
              children: "  description"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"Retrieve all user wallets"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " "
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "/>"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "snippet-categories",
      children: ["Snippet Categories", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#snippet-categories",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Organize your snippets into logical categories:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "docs/"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "└── snippets/"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    ├── setup/"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    │   ├── wallet-setup.mdx"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    │   └── api-setup.mdx"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    ├── errors/"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    │   ├── common-errors.mdx"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    │   └── error-handling.mdx"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    └── examples/"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        ├── react-example.mdx"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "        └── node-example.mdx"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "best-practices",
      children: ["Best Practices", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#best-practices",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Keep snippets atomic"
        }), " - Design snippets for a single, specific purpose"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Maintain consistency"
        }), " - Use consistent naming and formatting"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Update centrally"
        }), " - When changes are needed, update the snippet once"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Document parameters"
        }), " - For parameterized snippets, document expected parameters"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Version control"
        }), " - Keep snippets under version control along with your documentation"]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
