// vocs/_lib/vite/.vocs/dist/assets/walletconnect-B0NWUYgm.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Intersend App Store Technical Documentation",
  "description": "Start building your app on Intersend"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h2: "h2",
    h3: "h3",
    h4: "h4",
    li: "li",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsxs(_components.h2, {
      id: "brief-overview",
      children: ["Brief Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#brief-overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Intersend App Store is a non-custodial app marketplace that provides users with modular applications connected through a unified wallet."
    }), "\n", jsx(_components.p, {
      children: "Developers can leverage Intersend APIs, Iframe, and, soon, an SDK to obtain necessary permissions and connect to users' dedicated wallets. Intersend facilitates easy transactions for both Web3 and Web2 users, enabling them to send or receive any cryptocurrency seamlessly and onboard into the ecosystem efficiently."
    }), "\n", jsxs(_components.h2, {
      id: "main-use-cases",
      children: ["Main Use Cases", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#main-use-cases",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Easy Onboarding"
        }), ": Simplifies the process for users to make single-click payments by connecting their wallets."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Unified Wallet Management"
        }), ": Centralized control and management of crypto assets through a single wallet interface."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Omnichain Compatibility"
        }), ": Supports multiple blockchain networks, enabling seamless cross-chain transactions."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Gas-Free Transactions"
        }), ": Allows certain transactions to be conducted without gas fees, improving user experience."]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "ways-to-integrate",
      children: ["Ways to Integrate", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ways-to-integrate",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "1-your-web-application-hosted-by-yourself-such-as-iframe",
      children: ["1. Your Web Application Hosted by Yourself (such as iframe)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-your-web-application-hosted-by-yourself-such-as-iframe",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Your component (this could be an iframe or a webpage directly) will be hosted by you and accessed through the Intersend App Store."
    }), "\n", jsx(_components.p, {
      children: "The iframe sends events, such as making a transaction with details to the deposit address. We capture and process these events on the backend and send the response back to the developer's iframe or web app, which then displays the details to the frontend."
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: "User                  Developer's App (Accessed via Intersend)       Intersend Backend"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |-- Access App on Store ----------->|                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |- Send Event (e.g., transaction) ->|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |-- Process Event ------------------>|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |<-- Transaction Result -------------|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |<-- Receive Event Result ----------|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                                   |                                   |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |<-- Display Result ----------------|                                   |"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "2-intersend-iframe-integration",
      children: ["2. Intersend Iframe Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-intersend-iframe-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "You can utilize our iframe component that you can add to your website."
    }), "\n", jsx(_components.p, {
      children: "Our iframe is a light version of Intersend. It essentially provides a button that opens a pop-up window where users can log in, see their balance, approve transactions, and more."
    }), "\n", jsx(_components.p, {
      children: "For each event, your app and our app will constantly share these events with each other and execute relevant functions. These events include obtaining certain permissions, creating payment records, processing transactions, and more."
    }), "\n", jsxs(_components.h4, {
      id: "example-code-snippet",
      children: ["Example code snippet", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#example-code-snippet",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "### Intersend Iframe Integration Snippet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "```html"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "<iframe"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  width="375"'
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  height="667"'
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  src="https://embed.intersend.io/?ref=your_referral_id&utm_source=intersend_demo&theme=your_theme"'
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '  sandbox="allow-same-origin allow-popups allow-scripts allow-forms"'
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "/>"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "<script>"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  window.onmessage = function(e) {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    if (e.origin !== 'https://embed.intersend.io') {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      return;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    const {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      event,"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      data"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    } = e.data;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    switch(event) {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      case 'request_permission':"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        // Handle permission request with `"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "data"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "`"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        break;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      case 'payment_intent':"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        // Handle payment intent with `"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "data"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "`"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        break;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      case 'create_user':"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        // Handle user creation with `"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "data"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "`"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        break;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      case 'generate_wallet_address':"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        // Handle wallet address generation with `"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#DCBDFB"
              },
              children: "data"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "`"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        break;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      default:"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "        break;"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  // Example of sending an event to the iframe"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  const iframe = document.querySelector('iframe');"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  iframe.contentWindow.postMessage({"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    event: 'init',"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    data: {"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      userId: 'your_user_id',"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "      sessionId: 'your_session_id'"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "  }, 'https://embed.intersend.io');"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "<\/script>"
            })
          })]
        })
      })
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: "User                   Web App                Intersend Iframe            Intersend Wallet"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |-- Access Web App ----->|                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |-- Open Iframe --------->|                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |-- Log In User ----------->|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |<-- Display Balance -------|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |-- Approve Transaction --->|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |<-- Share Event -----------|"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                         |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |<-- Transaction Result--|                         |                           |"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "apis",
      children: ["APIs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#apis",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "With our APIs, you can create user wallets and enable them to receive any cryptocurrency on any chain, including EVM, Bitcoin, Solana, and many others."
    }), "\n", jsxs(_components.h3, {
      id: "create-user-wallet",
      children: ["Create User Wallet", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#create-user-wallet",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.pre, {
      children: jsx(_components.code, {
        className: "language-code",
        children: `curl -X POST https://node.intersend.io/v1/appstore/create-user \\
-H "Authorization: Bearer authToken" \\
-H "Content-Type: application/json" \\
-d '{
  "email": "user@example.com"
  // or "walletAddress"
}'
`
      })
    }), "\n", jsx(_components.p, {
      children: "This endpoint returns a link for the user to log in to the Intersend dashboard."
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: "Developer              Intersend API              Intersend Database"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |-- Create User Request->|                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |<-- User Login Link ----|                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |                           |"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              children: " |                        |-- Store User Data ------->|"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "generate-wallet-address",
      children: ["Generate Wallet Address", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#generate-wallet-address",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.pre, {
      children: jsx(_components.code, {
        className: "language-code",
        children: `curl -X POST https://node.intersend.io/v1/appstore/generate-wallet-address \\
-H "Authorization: Bearer authToken" \\
-H "Content-Type: application/json" \\
-d '{
 "email": "user@example.com",
 "chain": "BTC",
 "crypto": "BTC"
}'
`
      })
    }), "\n", jsx(_components.p, {
      children: "This endpoint returns a wallet address of the selected currency and a payment URI. If the user deposits funds to this address, their Intersend account will be deposited with the equivalent amount of USDT."
    }), "\n", jsx(_components.p, {
      children: "The user can sign into their Intersend account and access the funds."
    }), "\n", jsxs(_components.h2, {
      id: "event-types",
      children: ["Event Types", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#event-types",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "request_permission",
      children: ["request_permission", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#request_permission",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "This is the very initial step your app needs to ask the user. Typically, this is a one-time event for each user depending on your use case. You'd pass some security parameters such as:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "max_limit"
        }), ": The maximum amount limit your app will use."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "access_expiry"
        }), ": The expiration time for the access."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "user_wallet_address"
        }), ": The wallet address of the user."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "domain_whitelist"
        }), ": The list of domains that are allowed to access the permissions. (Your originating domain)"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "payment_intent",
      children: ["payment_intent", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#payment_intent",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "You can pass the information for payment such as the target wallet, amount, chain, and cryptocurrency (chain-agnostic). We then validate your permissions, process the transaction, and send the details through an event callback."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "target_wallet"
        }), ": The wallet address where the payment will be sent."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "amount"
        }), ": The amount of cryptocurrency to be sent."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "chain"
        }), ": The blockchain network (e.g., Ethereum, Bitcoin)."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "crypto"
        }), ": The type of cryptocurrency (e.g., BTC, ETH)."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "create_user",
      children: ["create_user", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#create_user",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Initiate the process to create a new user within the Intersend ecosystem. This event sets up the necessary user profile."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "email"
        }), ": The email address of the user."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "wallet_address"
        }), ": The wallet address of the user (optional if email is provided)."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "generate_wallet_address",
      children: ["generate_wallet_address", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#generate_wallet_address",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Generate a new wallet address for the user on a specified blockchain. This event returns a wallet address and a payment URI."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "email"
        }), ": The email address of the user."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "chain"
        }), ": The blockchain network for the wallet (e.g., BTC, ETH)."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "crypto"
        }), ": The type of cryptocurrency for the wallet (e.g., BTC, ETH)."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "transaction_status",
      children: ["transaction_status", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#transaction_status",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Check the status of a previously initiated transaction. This event provides updates on whether the transaction was successful, pending, or failed."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "transaction_id"
        }), ": The unique identifier of the transaction."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "revoke_permission",
      children: ["revoke_permission", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#revoke_permission",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Revoke previously granted permissions from a user. This event can be used to ensure security and compliance with user requests."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "user_wallet_address"
        }), ": The wallet address of the user."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.code, {
          children: "permission_id"
        }), ": The identifier of the permission to be revoked."]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "These events provide a comprehensive set of interactions that developers can use to build robust applications within the Intersend App Store ecosystem."
    }), "\n", jsxs(_components.h3, {
      id: "smart-contracts",
      children: ["Smart Contracts", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#smart-contracts",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "We use ERC-4337 smart contracts to facilitate secure and efficient transactions within the Intersend ecosystem."
    }), "\n", jsxs(_components.h2, {
      id: "contact-information",
      children: ["Contact Information", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#contact-information",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["For further questions, please reach out to ", jsx(_components.a, {
        href: "mailto:arda@intersend.io",
        children: "arda@intersend.io"
      }), " or @erturkarda on Telegram."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
